pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -161,26 +161,13 @@ func (cgc *containerGC) removeOldestNSandboxes(sandboxes []sandboxGCInfo, toRemo
 	// Remove from oldest to newest (last to first).
 	for i := len(sandboxes) - 1; i >= numToKeep; i-- {
 		if !sandboxes[i].active {
-			cgc.removeSandbox(sandboxes[i].id)
+			if err := cgc.manager.RemoveSandbox(sandboxes[i].id); err != nil {
+				klog.ErrorS(err, "Failed to remove sandbox", "sandboxID", sandboxes[i].id)
+			}
 		}
 	}
 }
 
-// removeSandbox removes the sandbox by sandboxID.
-func (cgc *containerGC) removeSandbox(sandboxID string) {
```

Comment:
Why isn't this change just to make `removeSandbox` here return an error?  That makes this a much smaller change (you could just change containerGC and make this a 4-10 line PR instead of 400).